### PR TITLE
Feat/dataprotection-target-location -- Initial TF resources & data source

### PR DESCRIPTION
**Pull request by GilTeraSky**
1. **What this PR does / why we need it**:
       This PR includes the Terraform resources for creating a target location.

2. **Which issue(s) this PR fixes**
        -

3. **Additional information**
       -

4. **Special notes for your reviewer**
       -

### CODE REVIEW

File: internal/resources/targetlocation/resource_target_location.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+		return diag.FromErr(errors.Wrap(err, "Couldn't create Tanzu Mission Control backup target location."))
+	}
+
+	err = validateSchema(model, credentialsType)
+
+	if err != nil {
+		return diag.FromErr(errors.Wrap(err, "Couldn't create Tanzu Mission Control backup target location."))
+	}
+
+	if model.FullName.Name == "gil-test2" {
```
Comment (rnarenpujari) [inline internal/resources/targetlocation/resource_target_location.go:56]: Maybe drop this?

File: internal/resources/targetlocation/resource_target_location.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+		if tlModel.Spec.Config.S3Config != nil && tlModel.Spec.Config.AzureConfig != nil {
+			return errors.New("Config should be set with either AWS or Azure blocks but not both.")
+		}
+	}
+
+	return nil
+}
+
+func modifyResponse(tlModel *targetlocationmodels.VmwareTanzuManageV1alpha1DataprotectionProviderBackuplocationBackupLocation, credentialsType credentialsmodels.VmwareTanzuManageV1alpha1AccountCredentialProvider) {
```
Comment (rnarenpujari) [inline internal/resources/targetlocation/resource_target_location.go:225]: nit: maybe add a comment explaining why this is needed?

**Review by rnarenpujari — CHANGES_REQUESTED** [review]

File: internal/resources/targetlocation/resource_target_location.go (hunk 2 above)
Comment (GilTeraSky) [reply internal/resources/targetlocation/resource_target_location.go:221]: Added a comment

File: internal/resources/targetlocation/resource_target_location.go (hunk 1 above)
Comment (GilTeraSky) [reply internal/resources/targetlocation/resource_target_location.go]: removed

**Review by rnarenpujari — APPROVED** [review]

**Pull request merged**